app/models/student.rb

```diff
@@ -156,8 +154,29 @@ def submission_for(code_review)
     submissions.find { |submission| submission.code_review_id == code_review.id }
   end
 
+  def pair_ids(course=nil)
+    selected_attendance_records = course ? attendance_records.where("date between ? and ?", course.start_date, course.end_date) : attendance_records
+    (selected_attendance_records.map {|s| s.pairings.pluck(:pair_id)}.flatten).compact.sort
+  end
+
   def pairs_on_day(day)
-    Student.where(id: attendance_record_on_day(day).try(:pair_ids))
+    Student.where(id: attendance_record_on_day(day).try(:pairings).try('pluck', 'pair_id'))
+  end
+
+  def inverse_pairs_on_day(day)
+    Student.where(id: AttendanceRecord.where(date: day).joins(:pairings).where(pairings: { pair_id: id }).pluck(:student_id))
+  end
+
+  def orphan_pairs_on_day(day) # extra pairs this student nonreciprocally claimed
+    pairs_on_day(day).where.not(id: inverse_pairs_on_day(day))
+  end
+
+  def inverse_orphan_pairs_on_day(day) # students who nonreciprocally claimed this student as a pair
+    inverse_pairs_on_day(day).where.not(id: pairs_on_day(day))
+  end
+
+  def pairs_without_feedback_today
+    pairs_on_day(Time.zone.now.to_date).where.not(id: evaluations_of_pairs.today.pluck(:pair_id))
```

Comment:
All these methods above refer to "pairs", but they are returning "paid_ids".

Reply:
These methods do return students actually: 
pairs_on_day
inverse_pairs_on_day
orphan_pairs_on_day
inverse_orphan_pairs_on_day
pairs_without_feedback_on_day

Reply:
Well gosh, I just saw the `pluck(:pair_id)` and didn't notice the double closing parenthesis. Oops sorry!